src/OpenTelemetry/Trace/Export/BatchingActivityProcessor.cs

```diff
@@ -87,18 +89,30 @@ public BatchingActivityProcessor(ActivityExporter exporter, int maxQueueSize, Ti
                 throw new ArgumentOutOfRangeException(nameof(maxExportBatchSize));
             }
 
+            if (scheduledDelay <= TimeSpan.FromMilliseconds(0))
+            {
+                throw new ArgumentOutOfRangeException(nameof(scheduledDelay));
+            }
+
             this.exporter = exporter ?? throw new ArgumentNullException(nameof(exporter));
             this.maxQueueSize = maxQueueSize;
             this.scheduledDelay = scheduledDelay;
             this.exporterTimeout = exporterTimeout;
             this.maxExportBatchSize = maxExportBatchSize;
 
-            this.cts = new CancellationTokenSource();
             this.exportQueue = new ConcurrentQueue<Activity>();
 
-            // worker task that will last for lifetime of processor.
-            // Threads are also useless as exporter tasks run in thread pool threads.
-            Task.Run(() => this.Worker(this.cts.Token), this.cts.Token);
+            this.flushTimer = new System.Timers.Timer
+            {
+                AutoReset = false,
```

Comment:
What does this mean? Timer.Elapsed is triggered only once? How does it keep getting trigger at every Interval?

Reply:
When the flush fires, the ~first thing it does is disable the timer:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/8ec6e7928d2a2034b663bea57ea8fbb042d844a6/src/OpenTelemetry/Trace/Export/BatchingActivityProcessor.cs#L195-L201

When it completes its work, it turns it back on:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/8ec6e7928d2a2034b663bea57ea8fbb042d844a6/src/OpenTelemetry/Trace/Export/BatchingActivityProcessor.cs#L241-L247

Let me know what you think about that as a mechanism.